cwms-data-api/src/main/java/cwms/cda/api/ForecastInstanceController.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    protected DSLContext getDslContext(Context ctx) {
+        return JooqDao.getDslContext(ctx);
+    }
+
+    @OpenApi(
+            description = "Used to create and save a forecast instance",
+            requestBody = @OpenApiRequestBody(
+                    content = {
+                            @OpenApiContent(from = ForecastInstance.class, type = Formats.JSONV2)
+                    },
+                    required = true
+            ),
+            method = HttpMethod.POST,
+            path = "/forecast-instance",
+            tags = TAG
+    )
+    @Override
+    public void create(@NotNull Context ctx) {
+        try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+            throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        }
+    }
+
+    @OpenApi(
+            description = "Used to delete forecast instance data based on unique fields",
+            queryParams = {
+                    @OpenApiParam(name = FORECAST_DATE_TIME, required = true, description = "Specifies the " +
+                            "owning office of the forecast instance to be deleted."),
+                    @OpenApiParam(name = ISSUE_DATE_TIME, required = true, description = "Specifies the " +
+                            "owning office of the forecast instance to be deleted."),
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the " +
+                            "owning office of the forecast spec associated with the forecast instance" +
+                            "to be deleted."),
+                    @OpenApiParam(name = SPEC_ID, required = true, description = "Specifies the " +
+                            "spec id of the forecast spec associated with the forecast instance" +
+                            "to be deleted."),
+                    @OpenApiParam(name = LOCATION, required = true, description = "Specifies the " +
+                            "location of the forecast spec associated with the forecast instance" +
+                            "to be deleted."),
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                            + "parameters did not find a forecast instance."),
+            },
+            path = "/forecast-instance",
+            method = HttpMethod.DELETE,
+            tags = TAG
+    )
+    @Override
+    public void delete(@NotNull Context ctx, @NotNull String forecastSpecId) {
+        try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+            throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        }
+    }
+
+    @OpenApi(
+            description = "Used to get all forecast instances for a given forecast spec",
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the " +
```

Comment:
None of the parameters are required in the getAll as they are just filters. Defaults would be unfiltered.

Reply:
I'm assuming this is true for the forecast spec get all too?